app/models/post.rb

```diff
@@ -57,6 +57,34 @@ def address
     [street, city, state, postal_code, country].compact.join(", ")
   end
 
+  def self.filter_type(type)
+    if type == "needed" || type == "available"
+      send(type.to_sym)
+    else
+      all
+    end
+  end
+
+  def self.filter_near(location)
+    results = near(location)
+
+    if results.blank?
```

Comment:
Yea, here I was filtering based on the *results* were blank, I should've been checking if the *query* was blank.